crypto/x509/x_pubkey.c

```diff
@@ -86,12 +86,14 @@ static void x509_pubkey_ex_free(ASN1_VALUE **pval, const ASN1_ITEM *it)
 {
     X509_PUBKEY *pubkey = (X509_PUBKEY *)*pval;
 
-    X509_ALGOR_free(pubkey->algor);
-    ASN1_BIT_STRING_free(pubkey->public_key);
-    EVP_PKEY_free(pubkey->pkey);
-    OPENSSL_free(pubkey->propq);
-    OPENSSL_free(pubkey);
-    *pval = NULL;
+    if (pubkey != NULL) {
+      X509_ALGOR_free(pubkey->algor);
+      ASN1_BIT_STRING_free(pubkey->public_key);
+      EVP_PKEY_free(pubkey->pkey);
+      OPENSSL_free(pubkey->propq);
+      OPENSSL_free(pubkey);
+      *pval = NULL;
+    }
```

Comment:
All lines in the body must be indented with two more spaces

Reply:
Another way would have been to just add these lines at the beginning:

``` C
    if (pubkey == NULL)
        return;
```

then you don't need to deal with re-indenting the whole body

Reply:
And testing for NULL at the beginning is the way all other openssl "free" routines are done.